src/rgen/Microsoft.Macios.Bindings.Analyzer/Validators/ClassValidator.cs

```diff
@@ -318,6 +318,40 @@ bool ValidAsyncMethods (Binding binding, RootContext context,
 		return diagnostics.Length == 0;
 	}
 
+	bool ValidProtocolInlineConstructors (Binding binding, RootContext context,
+		out ImmutableArray<Diagnostic> diagnostics, Location? location = null)
+	{
+		diagnostics = [];
+		// ensure that if there are any constructors that are going to be inlined from the protocols that they
+		// do not conflict with the constructors that are already defined in the class. This is a warning, and we only
+		// are about those constructors that have the same selectors. The user can disable the warning if he really has
+		var builder = ImmutableArray.CreateBuilder<Diagnostic> ();
+		// get all the selectors from the constructors defined in the class as well as the protocol ones and find
+		// the duplicates
+		var constructorSelectorsSet = binding.Constructors.ToDictionary (x => x.Selector!, x => x);
+		var duplicates = binding.ProtocolConstructors
+			.Where (x => x.Selector is not null && constructorSelectorsSet.ContainsKey (x.Selector))
+			.Select (x => x.Selector!)
+			.ToArray ();
+		if (duplicates.Length > 0) {
+			// we have duplicates, create a warning for each of them
+			foreach (var selector in duplicates) {
+				// use the class constructor location
+				var constructorLocation = constructorSelectorsSet.TryGetValue (selector, out var constructor)
+					? constructor.Location : location;
+				builder.Add (Diagnostic.Create (
+					descriptor: RBI0041, // The class '{0}' contains a constructor with the selector '{1}' that hides a inline constructor from a protocol
```

Comment:
Could you include the name/location of the protocol that brings in the conflicting constructor?